plutus-ledger-constraints/src/Ledger/Constraints/OnChain.hs

```diff
@@ -104,7 +104,8 @@ checkTxConstraint ctx@ScriptContext{scriptContextTxInfo} = \case
         $ isJust (V.findTxInByTxOutRef txOutRef scriptContextTxInfo)
     MustMintValue mps _ tn v ->
         traceIfFalse "L9" -- "Value minted not OK"
-        $ Value.valueOf (txInfoMint scriptContextTxInfo) (Value.mpsSymbol mps) tn == v
+        -- We add zero ada here to be consistent with cardano-ledger behaviour.
```

Comment:
This is the same comment as `OffChain.hs`. How about creating a [note](https://github.com/input-output-hk/plutus-apps/blob/main/CONTRIBUTING.adoc#notes) at the bottom of `OnChain.hs` and then refering to this note?

Reply:
Great idea! Thanks!

Reply:
Good point. I would say, why not? However, since there's no specification for how to construct the `ScriptContext`, this check feels ad-hoc.